sequence_task_3/student/src/impl.cpp

```diff
@@ -1,2 +1,26 @@
 #include "impl.h"
+#include <algorithm>
+
+std::string reverseWords(const std::string& input)
+{
+    std::stringstream istream(input);
+    std::deque<std::string> deque;
+
+    std::for_each(std::istream_iterator<std::string>(istream), std::istream_iterator<std::string>(), [&deque]( const std::string& s )
```

Comment:
Prefer more appropriate algorithm rather than std::for_each.
For example, std::copy is much suitable here.